src/javase/java/com/luciad/imageio/webp/WebPWriter.java

```diff
@@ -83,13 +83,22 @@ public void write(IIOMetadata streamMetadata, IIOImage image, ImageWriteParam pa
       throw new NullPointerException("Image may not be null");
     }
 
-    boolean encodeAlpha = hasTranslucency(aImage);
-    if (encodeAlpha) {
-      byte[] rgbaData = getRGBA(aImage);
-      return WebP.encodeRGBA(aOptions, rgbaData, aImage.getWidth(), aImage.getHeight(), aImage.getWidth() * 4);
-    } else {
-      byte[] rgbData = getRGB(aImage);
-      return WebP.encodeRGB(aOptions, rgbData, aImage.getWidth(), aImage.getHeight(), aImage.getWidth() * 3);
+    ThreadLocal<WebPEncoderOptions> encoderThreadLocal = new ThreadLocal<>();
```

Comment:
Why is ThreadLocal here preferred over any other reference-holding container, e.g. a WebPEncoderOptions[] ?

And wouldn't calling an aOptions.noopForAvoidingGC() method in the finally block have the same effect, since it prevents removing the object until then? This would spare us creating the extra container object.